opencog/guile/SchemeSmobTV.cc

```diff
@@ -460,5 +460,32 @@ SCM SchemeSmob::ss_tv_get_value (SCM s)
 	return SCM_EOL;
 }
 
+/**
+ * Return the truth value mean
+ */
+SCM SchemeSmob::ss_tv_get_mean(SCM s)
+{
+	TruthValue *tv = verify_tv(s, "cog-tv-mean");
+	return scm_from_double(tv->getMean());
+}
```

Comment:
Don't confuse memoizng and compiling.  Memoizing simply means that all lambdas have been expanded, or whatever. Don't know the details.  Compiling is much much more: the expressions are converted into byte-code, which is then cached in the file-system.  The byte-code runs at typical VM speeds -- comparable to Java I suppose, which is typically 1/3rd the speed of native assembly code.  I have not measured.  By default, all scheme code is compiled, so tv->alist would be compiled, and presumably the optimizer would unwind and unroll references & etc.  whatever it does ... there is a way to turn off compilation and memoization and always run  interpreted, and I did measure that once for opencog, it wasn't horrible but it wan't great.  (It wasn't horrible mostly because the guile snippets in opencog are always rather small and short.  If we were writing giant programs in guile, it would be a huge difference. But most code in e.g. utilities.scm is small.)